Code/Tools/AssetProcessor/native/tests/assetmanager/IntermediateAssetTests.cpp

```diff
@@ -782,4 +782,56 @@ namespace UnitTests
         ProcessSingleStep();
         CheckProduct("test.stage4"); // Same product result at the end
     }
+
+    TEST_F(IntermediateAssetTests, UpdateSource_OutputDoesntChange_IntermediateDoesNotReprocess)
```

Comment:
what actually checks for "OutputDoesntChange_IntermediateDoesNotReproces" ?

Reply:
IntermediateDoesNotReprocess is the goal but probably not the most direct way to describe what we're testing.  IntermediateJobIsNotEmitted is probably more accurate.

The last line of the test calls RunFile and passes in the expected number of jobs we should see emitted (0).  I've updated RunFile to make sure its always checking the final job count, since it was avoiding that check in some cases.